[Security] Remove unnecessary inherited doc annotation

junaidbinfarooq opened the pull request:
Removes unnecessary inherited doc annotation from php doc from the methods that are not inherited from any parent class/interface.
Adds Doc blocks from the relevant methods in the `AdvancedUserInterface` (Removed in 5.0).

| Q             | A
| ------------- | ---
| Branch?       | 5.2
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no
| Tickets       | Fix #... <!-- prefix each issue number with "Fix #", no need to create an issue if none exist, explain below instead -->
| License       | MIT
| Doc PR        | symfony/symfony-docs#... <!-- required for new features -->
<!--
Replace this notice by a short README for your feature/bugfix. This will help people
understand your PR and can be used as a start for the documentation.

Additionally (see https://symfony.com/releases):
 - Always add tests and ensure they pass.
 - Never break backward compatibility (see https://symfony.com/bc).
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against branch 5.x.
 - Changelog entry should follow https://symfony.com/doc/current/contributing/code/conventions.html#writing-a-changelog-entry
-->

junaidbinfarooq committed:
[Security] Remove unnecessary inherited doc annotation

Nyholm reviewed (approved):
These method was added in 2010. It is true that they are not covered by any interface.

derrabus commented:
> These method was added in 2010. It is true that they are not covered by any interface.

The interface has been removed in 5.0.

derrabus commented:
I'm fine with this change, however I think we should copy the doc blocks from the old interface instead of just deleting the `@inheritdoc`s:

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Security/Core/User/AdvancedUserInterface.php

Also, the change should be done on the 5.2 branch.

junaidbinfarooq commented:
> I'm fine with this change, however I think we should copy the doc blocks from the old interface instead of just deleting the `@inheritdoc`s:
> 
> https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Security/Core/User/AdvancedUserInterface.php
> 
> Also, the change should be done on the 5.2 branch.

Thanks for the review.
So should I go ahead with adding the doc blocks from `AdvancedUserInterface` to this class or leave it as such?
And regarding your last point, should I mention 5.2 in the QA in the initial comment?

Nyholm commented:
> So should I go ahead with adding the doc blocks from AdvancedUserInterface to this class

Yes please. 

> And regarding your last point, should I mention 5.2 in the QA in the initial comment?

Sure, you can update the PR description. But more importantly, click on the "Edit" button next to the PR title and change the base to 5.2.

wouterj commented:
> . But more importantly, click on the "Edit" button next to the PR title and change the base to 5.2.

In case you don't know, this also requires re-basing your branch from 5.x to 5.2 using git:

```
# given "upstream" is the git remote pointing to symfony/symfony, this rebases from the
# symfony/symfony:5.x branch onto the symfony/symfony:5.2 branch (removing the 479
# unrelated commits from the PR diff)
git rebase --onto upstream/5.2 upstream/5.x
```

junaidbinfarooq commented:
> > So should I go ahead with adding the doc blocks from AdvancedUserInterface to this class
> 
> Yes please.
> 
> > And regarding your last point, should I mention 5.2 in the QA in the initial comment?
> 
> Sure, you can update the PR description. But more importantly, click on the "Edit" button next to the PR title and change the base to 5.2.

Thanks.
Changed the base to 5.2. It apparently lead to problems with tons of commits from the other branch showing up in this branch. I rebased the said branch on 5.2 and force pushed the changes but that didn't help either.
I added the relevant doc blocks though.

derrabus commented:
Let me fix that for you.

derrabus commented:
Good catch, thanks @junaidbinfarooq.

Nyholm commented:
> I rebased the said branch on 5.2 and force pushed the changes but that didn't help either.

Sometimes git does not play well with me either. Lucky @derrabus is so helpful. 

Thank you @junaidbinfarooq! Congratulations to your first contribution 🎉

junaidbinfarooq commented:
> Let me fix that for you.

Thank you so much.

junaidbinfarooq commented:
> > I rebased the said branch on 5.2 and force pushed the changes but that didn't help either.
> 
> Sometimes git does not play well with me either. Lucky @derrabus is so helpful.
> 
> Thank you @junaidbinfarooq! Congratulations to your first contribution

Thank you so much :smile:. Hope I can contribute more in the future.

junaidbinfarooq commented:
> > . But more importantly, click on the "Edit" button next to the PR title and change the base to 5.2.
> 
> In case you don't know, this also requires re-basing your branch from 5.x to 5.2 using git:
> 
> ```
> # given "upstream" is the git remote pointing to symfony/symfony, this rebases from the
> # symfony/symfony:5.x branch onto the symfony/symfony:5.2 branch (removing the 479
> # unrelated commits from the PR diff)
> git rebase --onto upstream/5.2 upstream/5.x
> ```

Thanks for the info.
I did that, though via PHPStorm, and then force-pushed the changes but there were still many commits showing up in the PR. Not sure why so?

derrabus commented:
Because PhpStorm does not guess your branch point and thus performs the rebase without the `--onto` parameter.

junaidbinfarooq commented:
Strange. It usually works with the codebase I work with.
